### PR TITLE
RAC-6635 Adding events to Redfish discovery

### PR DESCRIPTION
**Pull request by BrentHiggins**
@jimturnquist @larry-dean

### CODE REVIEW

**Comment by BrentHiggins** [comment]
Fixed a bug where one of the events would be published for the wrong node

**Comment by BrentHiggins** [comment]
Moved to WIP due to bug found

File: lib/jobs/general-redfish-catalog.js [hunk 1]
```diff
@@ -162,9 +171,32 @@ function GeneralRedfishCatalogJobFactory(
...
+                    .then(function(catalog) {
+                        if (!_.isEqual(catalog[0].data, dataToSave.data)) {
+                            return waterline.catalogs.update(query, dataToSave);
+                        }
+                    });
                 } else {
-                    return waterline.catalogs.create(dataToSave);
+                    return waterline.catalogs.create(dataToSave)
+                    .then(function() {
+                        return waterline.catalogs.update(query, dataToSave);
```
Comment (BrentHiggins) [inline lib/jobs/general-redfish-catalog.js]: We are doing the update to get the "."'s back in the odata keys.
This fixes the entries back to what is expected for redfish data.

**Comment by BrentHiggins** [comment]
We have fixed the bug and changed that catalogs are only modified if changes were found

**Comment by BrentHiggins** [comment]
@lanchongyizu we have made changes if you are interested

File: lib/jobs/general-redfish-catalog.js [hunk 2]
```diff
@@ -124,7 +126,7 @@ function GeneralRedfishCatalogJobFactory(
 
                 default:
                     logger.info("Node type: " + node.type);
-                    return;
+                    return ;
```
Comment (jimturnquist) [inline lib/jobs/general-redfish-catalog.js]: Minor formatting nit here.  Extra space

**Comment by lanchongyizu** [comment]
@BrentHiggins LGTM

**Pull request merged**